apps/chat/src/utils/app/application.ts

```diff
@@ -61,13 +60,22 @@ interface ApplicationDetailsResponse {
   reference: string;
 }
 
+interface ApplicationDetailsResponse extends BaseApplicationDetailsResponse {
+  name: string;
+}
+
+interface PublicApplicationDetailsResponse
+  extends BaseApplicationDetailsResponse {
+  application: string;
+}
+
 export const convertApplicationFromApi = (
-  application: ApplicationDetailsResponse,
+  application: ApplicationDetailsResponse | PublicApplicationDetailsResponse,
 ): CustomApplicationModel => ({
   ...application,
   isDefault: false,
   type: EntityType.Application,
-  id: application.name,
+  id: 'name' in application ? application.name : application.application,
```

Comment:
if `application` exists only in public-applications then it should be opposite way:
```suggestion
  id: 'application' in application ? application.application: application.name,
```